cheatsheets_draft/GraphQL_Cheat_Sheet.md

```diff
@@ -0,0 +1,128 @@
+
+Implementing this should be very simple for most data types. Whitelist the characters that are allowed for any input given to the API and throw an error if illegal characters are given. For example, a user's ID might only contain numbers, so limiting an input field for user IDs to numbers is the best approach. The allowed characters will depend on the piece of data being passed. If alphanumeric characters are allowed there is very little risk of SSRF with an HTTP call. There is also very little risk for SQL injection, but any DB queries should be parameterized anyway. Unicode characters should also be disallowed unless they are specifically required in a field. If the field must have more than alphanumeric characters or Unicode then the other protection options (sanitization and/or parameterization) will be needed.
+
+This can be achieved by using specific GraphQL types (such as numbers or enums) or by writing custom GraphQL [validators](https://graphql.org/learn/validation/) for more complicated validations.  
+
+See the OWASP Cheat Sheet on [Input Validation](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html) for more info.
+
+#### SQL Injection Prevention
+
+Anytime a DB or similar query is being made, any input should be properly parameterized with prepared statements or stored procedures in order to prevent SQL/DB injection. See the [OWASP Cheat Sheet for SQL Injection Prevention](https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html) for more info.
+
+#### Process Validation
+
+When using user input, even if sanitized and/or validated it should not be used for certain purposes that would give a user control over data flow. For example, do not make an HTTP/resource request to a host that the user supplies (unless there is a business need).
+
+### Resource Management
+
+> I assume we don't want the "internal vs external" API talk in here?
+
+For a GraphQL API that receives input more or less directly from an external party, it can be difficult to limit the possibilities a malicious user can use to exhaust the API's resources to cause slow downs or indefinite outages. It is easier to protect an API that is only used by other internal services. Because of this, it may be better to only expose your API internally. See the table below for which of these controls should be added to each type of GraphQL API.
+
+#### Depth Limiting
+
+This is limiting the depth of incoming queries so that the user cannot supply unnecessarily deep queries that could consume a lot of resources. See [this](https://www.apollographql.com/blog/securing-your-graphql-api-from-malicious-queries-16130a324a6b) and [this blog post](https://www.howtographql.com/advanced/4-security/) as well as [this Github repo](https://github.com/stems/graphql-depth-limit) (old, potentially unsupported) for more details on implementing this control. I'm not sure if there is a good/trusted open source project that does this.
```

Comment:
Yeah from my quick searching I couldn't really find anything legit or trusted. We can def delete this. It seems like the code to do this is pretty simple though; would be cool to see an official OWASP project tackle this!

Reply:
If you feel that it is worth to have a short code snippet here we can think about it. 
We try to not have longer code samples because they are hard to maintain but short snippet or pseudocode is a good idea sometimes.